server/client.go

```diff
@@ -1501,6 +1506,18 @@ func (c *client) processConnect(arg []byte) error {
 	account := c.opts.Account
 	accountNew := c.opts.AccountNew
 
+	ncs := c.String()
```

Comment:
May I recommend something like this instead?
```
if c.kind == CLIENT {
        ncs := c.String()
	start := " - ["
	end := ""
	if c.opts.Version != "" {
		if c.opts.Name == _EMPTY_ && c.opts.Lang == _EMPTY_ {
			end = "]"
		}
		ncs = fmt.Sprintf("%s%sv%s%s", ncs, start, c.opts.Version, end)
		start = "-"
	}
	if c.opts.Name != "" {
		if c.opts.Lang == _EMPTY_ {
			end = "]"
		}
		ncs = fmt.Sprintf("%s%s%s%s", ncs, start, c.opts.Name, end)
		start = "-"
	}
	if c.opts.Lang != "" {
		end = "]"
		ncs = fmt.Sprintf("%s%sLang:%s%s", ncs, start, c.opts.Lang, end)
	}
	c.ncs.Store(ncs)
}
```
This would produce this kind of output:
```
[91713] 2020/09/11 09:24:36.326023 [TRC] 127.0.0.1:54335 - cid:2 - [v1.11.0-NATS Sample Publisher-Lang:go] - <<- [PING]
...
[91713] 2020/09/11 09:25:01.655146 [TRC] [::1]:54336 - cid:3 - [v2.2.0-Lang:C] - <<- [PING]
```
It is more compact and prevent the confusion that `lang: go - version: 1.11.0` may lead to believe that it is Go version 1.11.0.

Also, I would like to limit this to client for now, because otherwise, accepting routes would have the remote server name (which could be the pub key, which is rather long), etc..

Reply:
I like the idea of putting it within brackets since that makes it easier for processing the logs and also more concise. Although since this is arbitrary user metadata that ends up in the logs I think we will have escape and put within quotes to avoid log file injection issues which is one of the things that are checked during security reviews, e.g.

```
echo 'connect {"name":"\\n#!/bin/sh\\necho foo\\n"}' | nc 127.0.0.1 4222
```

Based on @kozlovic snippet I was thinking maybe something like this for the format? Thought maybe could change for this field to use `:` as well so that format is similar like with client ids (`cid:3`).

```
[64286] 2020/09/12 10:51:03.211459 [TRC] 127.0.0.1:55809 - cid:2 - "v1.9.1:NATS Sample Subscriber:go" - <<- [PING]
[64286] 2020/09/12 10:51:03.211517 [TRC] 127.0.0.1:55809 - cid:2 - "v1.9.1:NATS Sample Subscriber:go" - ->> [PONG]
[64286] 2020/09/12 10:51:03.211835 [TRC] 127.0.0.1:55810 - cid:3 - "v0.11.2:python3" - <<- [PING]
[64286] 2020/09/12 10:51:03.211881 [TRC] 127.0.0.1:55810 - cid:3 - "v0.11.2:python3" - ->> [PONG]
```

```go
	if c.kind == CLIENT {
		var ncs string
		if c.opts.Version != "" {
			ncs = fmt.Sprintf("v%s", c.opts.Version)
		}
		if c.opts.Name != "" {
			if c.opts.Version == _EMPTY_ {
				ncs = c.opts.Name
			} else {
				ncs = fmt.Sprintf("%s:%s", ncs, c.opts.Name)
			}
		}
		if c.opts.Lang != "" {
			if c.opts.Version == _EMPTY_ && c.opts.Name == _EMPTY_ {
				ncs = c.opts.Lang
			} else {
				ncs = fmt.Sprintf("%s:%s", ncs, c.opts.Lang)
			}
		}
		c.ncs.Store(fmt.Sprintf("%s - %q", c.String(), ncs))
	}
```

Reply:
Since libraries provide a version and lang, maybe move the optional name towards the end.